src/pybind/mgr/dashboard/services/sso.py

```diff
@@ -46,17 +48,19 @@ class SsoDB(object):
     VERSION = 1
     SSODB_CONFIG_KEY = "ssodb_v"
 
-    def __init__(self, version, protocol, saml2):
+    def __init__(self, version, protocol, saml2, oauth2):
         self.version = version
         self.protocol = protocol
         self.saml2 = saml2
+        self.oauth2 = oauth2
         self.lock = threading.RLock()
 
     def save(self):
         with self.lock:
             db = {
                 'protocol': self.protocol,
                 'saml2': self.saml2.to_dict(),
+                'oauth2': self.oauth2.to_dict(),
```

Comment:
Internally you could still save OAuth2 config to a new entry, like this, in order to make this DB schema backward-compatible.